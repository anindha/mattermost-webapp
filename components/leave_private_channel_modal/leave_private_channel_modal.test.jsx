// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {shallow} from 'enzyme';
import React from 'react';

import LeavePrivateChannelModal from 'components/leave_private_channel_modal/leave_private_channel_modal.jsx';
import Constants from 'utils/constants';

describe('components/LeavePrivateChannelModal', () => {
    const channels = {
        'channel-1': {
            id: 'channel-1',
            name: 'test-channel-1',
            display_name: 'Test Channel 1',
            type: 'P',
            team_id: 'team-1',
        },
        'channel-2': {
            id: 'channel-2',
            name: 'test-channel-2',
            display_name: 'Test Channel 2',
            type: 'P',
            team_id: 'team-1',
        },
        'town-square': {
            id: 'town-square-id',
            name: 'town-square',
            display_name: 'Town Square',
            type: 'O',
            team_id: 'team-1',
        },
    };

    const baseProps = {
        onHide: jest.fn(),
        show: false,
        channel: channels['channel-1'],
        actions: {
            leaveChannel: jest.fn(),
        },
    };

    test('should match snapshot, init', () => {
        const wrapper = shallow(
            <LeavePrivateChannelModal
                {...baseProps}
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    test('should fail to leave channel', (done) => {
        const props = {
            ...baseProps,
            show: true,
            channel: channels['channel-2'],
            actions: {
                leaveChannel: jest.fn().mockImplementation(() => {
                    const error = {
                        message: 'error leaving channel',
                    };

                    return Promise.resolve({error});
                }),
            },
        };
        const wrapper = shallow(
            <LeavePrivateChannelModal {...props}/>
        );

        const instance = wrapper.instance();
        instance.handleSubmit();
        expect(instance.props.actions.leaveChannel).toHaveBeenCalledTimes(1);
        process.nextTick(() => {
            expect(instance.props.onHide).toHaveBeenCalledTimes(0);
            done();
        });
    });

    test('should leave channel when pressing the enter key browse to default channel', (done) => {
        const props = {
            ...baseProps,
            show: true,
            channel: channels['channel-1'],
            actions: {
                leaveChannel: jest.fn().mockImplementation(() => {
                    const data = true;

                    return Promise.resolve({data});
                }),
            },
        };
        const wrapper = shallow(
            <LeavePrivateChannelModal {...props}/>
        );

        const instance = wrapper.instance();
        const enterKey = {
            preventDefault: jest.fn(),
            ctrlKey: false,
            key: Constants.KeyCodes.ENTER[0],
            keyCode: Constants.KeyCodes.ENTER[1],
        };

        instance.handleKeyPress(enterKey);
        expect(instance.props.actions.leaveChannel).toHaveBeenCalledTimes(1);
        process.nextTick(() => {
            expect(instance.props.onHide).toHaveBeenCalledTimes(1);
            done();
        });
    });
});
