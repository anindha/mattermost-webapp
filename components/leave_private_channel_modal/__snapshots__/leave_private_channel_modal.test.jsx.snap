// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/LeavePrivateChannelModal should match snapshot, init 1`] = `
<ConfirmModal
  confirmButtonClass="btn btn-danger"
  confirmButtonText={
    <FormattedMessage
      defaultMessage="Yes, leave channel"
      id="leave_private_channel_modal.leave"
      values={Object {}}
    />
  }
  message={
    <FormattedMessage
      defaultMessage="Are you sure you wish to leave the private channel {channel}? You must be re-invited in order to re-join this channel in the future."
      id="leave_private_channel_modal.message"
      values={
        Object {
          "channel": <b>
            Test Channel 1
          </b>,
        }
      }
    />
  }
  modalClass=""
  onCancel={[MockFunction]}
  onConfirm={[Function]}
  show={false}
  title={
    <FormattedMessage
      defaultMessage="Leave Private Channel {channel}"
      id="leave_private_channel_modal.title"
      values={
        Object {
          "channel": <b>
            Test Channel 1
          </b>,
        }
      }
    />
  }
/>
`;
